### PR TITLE
types: simplify custom TS types

### PR DESCRIPTION
**Pull request by eszthoff**
Story: [JF-3919](https://textkernel.atlassian.net/browse/JF-3919)

### CODE REVIEW

File: src/components/Buttons/FileButton/__tests__/FileButton.spec.tsx [hunk 1]
```diff
@@ -5,6 +5,11 @@ import userEvent from '@testing-library/user-event';
 import { FileButton } from '../FileButton';
 
 describe('<FileButton> that renders a button', () => {
+    it('should render nothing if no children are provided', () => {
+        const { container } = render(<FileButton>{}</FileButton>);
+
+        expect(container.firstChild).toBeNull();
```
Comment (andrentk) [inline src/components/Buttons/FileButton/__tests__/FileButton.spec.tsx:11]: jest provides toBeEmptyDOMElement that might be more self explanatory to use in this type of cases.

Comment (eszthoff) [reply src/components/Buttons/FileButton/__tests__/FileButton.spec.tsx:11]: good one. I was looking for it, but somehow didn't find it... Thanks

File: src/components/Heading/Heading.tsx [hunk 2]
```diff
@@ -19,8 +18,16 @@ const { block } = bem('Heading', styles);
 
 export const Heading = React.forwardRef<HTMLElement, Props>(
     ({ align = 'left', children, isNeutral, level = 'h1', ...rest }, ref) => {
+        if (!children) {
+            return null;
+        }
+
         const HtmlNodeType = level;
 
+        if (!children) {
```
Comment (andrentk) [inline src/components/Heading/Heading.tsx:27]: Duplicated if statement.

File: src/components/Icon/__tests__/IconBase.spec.tsx [hunk 3]
```diff
@@ -4,6 +4,11 @@ import '@testing-library/jest-dom';
 import { IconBase } from '../IconBase';
 
 describe('<IconBase> that renders an SVG wrapper with all options included', () => {
+    it('should render nothing if no children are provided', () => {
+        const { container } = render(<IconBase viewBox="0 0 100 100">{}</IconBase>);
+
+        expect(container.firstChild).toBeNull();
```
Comment (andrentk) [inline src/components/Icon/__tests__/IconBase.spec.tsx:10]: toBeEmptyDOMElement missing here.

**Review by andrentk — APPROVED** [review]

**Pull request merged**